Content.Server/Silicons/Laws/StartIonStormedSystem.cs

```diff
@@ -0,0 +1,40 @@
+using Content.Shared.Silicons.Laws.Components;
+using Content.Shared.Administration.Logs;
+using Content.Shared.Database;
+using Content.Shared.Silicons.Laws;
+
+namespace Content.Server.Silicons.Laws;
+
+/// <summary>
+/// This handles running the ion storm event on specific entities when spawned in.
+/// </summary>
+public sealed class StartIonStormedSystem : EntitySystem
+{
+    [Dependency] private readonly IonStormSystem _ionStorm = default!;
+    [Dependency] private readonly ISharedAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly SiliconLawSystem _siliconLaw = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<StartIonStormedComponent, MapInitEvent>(OnMapInit);
+    }
+
+    private void OnMapInit(EntityUid uid, StartIonStormedComponent component, ref MapInitEvent args)
+    {
+        if (!TryComp<SiliconLawBoundComponent>(uid, out var lawBound))
+            return;
+        if (!TryComp<TransformComponent>(uid, out var xform))
+            return;
+        if (!TryComp<IonStormTargetComponent>(uid, out var target))
+            return;
+
+        for (int currentIonStorm = 1; currentIonStorm <= component.IonStormAmount; currentIonStorm++)
```

Comment:
```suggestion
        for (int currentIonStorm = 0; currentIonStorm < component.IonStormAmount; currentIonStorm++)
```
Programmers start counting at 0 😄 
(your way is fine of course)